### ISSUE TITLE
Editing the Notebook

### ISSUE THREAD

**Issue by MadisonJMyers** [aloknsingh/ibm_r4ml_svm_classification#2]
Alok- great work! I think this is a very interesting idea and I can tell a lot of work has gone into this. You are almost there! There are some grammatical and formatting errors that I've listed below that you should take a look at and implement, but otherwise this is great.

My edits:

- I believe the journey team wants number formatting to have a period afterwards (1. , 2. , 3. etc.) Please change the numbering throughout the notebook to be in accordance with that.
- I assume this is a draft and that is why there are question marks after some of the headers. Be sure to remove those or rephrase!
- _“This section, we will walk through the general concept of model training. Use the R4ML to do the Support vector machine classification. We will train and score on the data and lastly we will do the cross validation to do the model tuning to select best set of hyperparameters.”_ should be changed to be: “In this section, we will walk through the general concept of model training and then use the R4ML to conduct Support vector machine classification. We will then train and score the data and conduct cross validation in order to select the best set of hyperparameters.”
- _“We talked in the part I and II , how to load R4ML and do some preprocessing on the data. We will use the boiler plate code from the previous sections. Do visit those section to go in details on each of these steps”_ is unclear. Please replace these lines with an explanation of what the notebook is doing below so that the reader can follow along.
- Again, most “? “ will need to be removed.
- _“Many real life problems and business involves analysising the data set and predicting certain outcomes. Basic statistics can help one with some of the insights but for deeper insights and predictive analytics, we need stronger and complex mathemathical models. There are various kind of machine learning models that are used. R4ML has builtin collection of many ML model training utils. Core R provides many ML models but the problem arises when the data set becomes huge and then we need scalable machine learning model. R4ML fills those gaps.”_ should be changed to: “Many real world problems  involve analyzing data and predicting certain outcomes. Basic statistics can help one with some of the insights, but to go deeper into predictive analytics, we need stronger and complex mathematical models. For this, there are various kind of machine learning models that are used. R4ML, for example, has a built-in collection of many ML models. Other packages offer ML models, however problems often arise when the data becomes huge and we need scalable machine learning models. R4ML is able to fill this gap.”
- _“To better understand the process using R4ML, we will first pose one prediction problem first.”_ should be: “To better understand this process using R4ML, we will begin by trying out an example prediction problem.”
- _“Given the other information about the flights, is it going to get delayed or not? A: The problem is to find if there is the delay or not. We note that there is the ArrDelay column, which is given to use. Lets assume that it is not available and we are trying to find the airline delay. Furthuer we will divide into categor no delay , less than 60 min delay, more than 60 min delay”_ should be deleted as it is unclear. You have already led the reader into the problem. 
- _“We saw in part II , a few data preparation steps. Here we will perform two extra pre-processing to increase the predictive power of the features.”_ should be changed to: “We saw a few data preparation points in step 2. Next we will perform two extra pre-processing steps to increase the predictive power of the features.”
- It would be better if you could refer to the parts as 1., 2., 3. etc to stay consistent rather than change it to “part II” and so on. This will make it easier for the reader to follow.
- _“because, if we don’t do it, then one variable may have a much bigger scale for example measured in seconds, and second variable measured in hours.”_ should be changed to: “Because, if we don’t do it, then one variable may have a much larger scale. For example the scale could be inconsistently measured in a mix of seconds and hours.”
- Put a period at the end of the sentences. Specifically after the sentence which ends “…the value has been normalized.”
- Please capitalize the beginning of your sentences. Specifically the sentence which starts out with “Next, we will…”
- That sentences needs corrected further to “Next, we will do encoding of the categorical variable of interests.” (Though this is unclear and should be rephrased if you can.)
- _“There are two types of categorical variable ordinal and nominal. ordinal variable are like the binned continous variable. For example the size of t-shirts, blood presure ranges etc. However nominal variables are for example states of US. various airline carrier, various airport. If we just do the recoding and no one hot encoding, the output will be erroneous, since for example for the airport, (SFO, JFK), we can assign (1,5) that gives more weight JFK, which it shouldn’t do. one hot encoding will map the above to (1,0), (0,1) and that by adding extra features, we will have be able to convert categorical nominal variables into something that can be consume by the ML algorithm for classification.”_ should be changed to: “There are two types of categorical variables: ordinal and nominal. Ordinal variables are like the binned continuous variable. Examples include the size of t-shirts, blood pressure ranges and more. However, nominal variable examples differ and include US’ states, airline carriers, airports, etc. By conducting encoding, we will map the above to (1,0), (0,1) and by adding extra features, we will convert categorical, nominal variables into something that can be used by the ML algorithm for classification.”
- _“Since we would like to illustrate the classification process. We would convert our independent variable i.e Y == AirDelay into categorical variable.”_ Should be changed to: “Since we would like to illustrate the classification process. We would convert our independent variable i.e Y == AirDelay into categorical variable.” should be changed to: “In order to illustrate the classification process, next we will convert our independent variable (i.e Y == AirDelay) into a categorical variable.”
- _“R4ML provides a various built scalable ML algorithm. So which algorithm to use? we have the choice of Support Vector Machine(SVM) or Mlogit (Multiclass logistic regression). For this example, we will use the SVM. As the first naive approach, we will try to use the canned SVM algorthm provided by the R4ML. Before that lets talk a little about SVM”_ should be changed to: “As we have learned, R4ML provides a library with built-in, scalable ML algorithms. But which algorithm should you use? You have the choice of a Support Vector Machine (SVM) or Mlogit (Multiclass logistic regression). For this example, we will use the SVM. As the first naive approach, we will try to use the canned SVM algorithm provided by the R4ML. Before that lets talk a little about SVM”
- _“It is the constrained is a constrained optimization problem where the margin is maximized subject to the constraint that it perfectly classifies the data i.e hard margin.”_ should be: “The SVM is a constrained optimization problem where the margin is maximized, subject to the constraint that it perfectly classifies the data i.e hard margin.”
- _“Why SVM is preferred?”_ Should be “Why SVM is preferred:” or “Why should SVM be preferred?”
- _“Compare to logistic regression…?”_ should be: “Compared to logistic regression, the SVM algorithm has advantages in terms of complexity. These include: ”
- Make sure each bullet point ends in a period. I mentioned this earlier but I want to repeat it.
- _“SVM is found to have better performance practically in most cases.”_  should be: “SVM is found to have better practical performance in most cases.”
- _“Classifier in SVM depends only …”_ should be changed to: “The SVM classifier depends only on a subset of points . Since we need to maximize the distance between the closest points of two classes (aka margin) we need to focus only on a subset of points, unlike logistic regression.”
- _“accuracy on the later”_ should be “accuracy on the latter”
- _“We will see the more detailed cross validation steps later but for now, lets follow the simpler steps, where we train the model and score on it”_ should be changed to: “We will see the more detailed cross validation steps later but for now, lets follow the simpler steps where we train the model and score it.”
- Under R4ML model training, delete the sentence: _“Once all the data preparation is done. “_ 
- Delete _“straight forward”_ from the next sentence and add a colon on the end. (“The next step is to train a scalable machine learning model as follows:”
- Under 4.3 it _“Lets”_ should be “Let’s”
- _“As we know, that in classification models, accuracy can be misleading.”_ should be “As we know, accuracy in classification models can be misleading.”
- Add A to the beginning of the sentence _“Confusion matrix…”_
- _“Lets see how does R4ML predictions confusion matrix look like. (Note that the following plot has been rotated on the middle vertical axis to follow R's ggplot).”_ should be: “Lets see how a R4ML confusion matrix looks. (Note that the following plot has been rotated on the middle vertical axis to follow R's ggplot).”
- _“We observe in the following plot that the TP and TN is very high and FP and FN are are low and we are more confident of our model than just the accuracy based analysis.”_ should be: “We observe in the following plot that the TP and TN is very high and FP and FN are low.  This means we can be more confident of our model than just the accuracy-based analysis.”
- _“We can also see the predicted scores, the higher the score the higher the margin and better or more confident we are of the prediction”_ should be: “We can also see the predicted scores. The higher the score, the higher the margin and the better or more confident we are of the prediction.”
- In 4.4 add a colon after _“we saw”_ (“In this subsection, we saw:”).
- In 4.4 add a period at the end of each bullet point.
- _“5 Model Tuning via cross validation.”_ should be: “5. Model-Tuning via Cross-Validation” (main headers like that do not need a period at the end)
- All _cross validation_ should be changed to cross-validation with a hyphen. 
- _“Many models have important parameters which cannot be directly estimated from the data. For example, in the SVM classifification problem, to avoid overfitting, we would like to tune it using the tuning parameter lambda.The question is what value of tuning parameter we should use?”_ should be: “Many models have important parameters which cannot be directly estimated from the data. For example, in the SVM classification problem, to avoid overfitting, it should be adjusted using the tuning parameter lambda. The question is: what should the value be?”
- In the next paragraph, _“candidates values”_ should be “candidate values”
- Use a colon after _“…in the following figure.”_ (“…in the following figure:”)
- Remove the space between the first and second bullet points for consistency. (In 5.1)
- In the second bullet point change _“bet- ter”_ to “better”.
- _“We could also use the k fold cross validation but we will simply use in this example 2 fold cross validation”_ should be: “While the the k-fold cross-validation could be applied here, we will use the 2-fold cross-validation.”
- I like that you have a summary! 
- Please add a colon after _“we saw”_ (In this section, we saw:”)
- Change _“saw”_ to “learned”.
- Capitalize the beginnings of the bullet points.
- _“Arrival delay of airline”_ should be “airline delays.”
- Remove _“Then”_ from the beginning of the last bullet point and remove _“the”_ before _“classic”_ so that is is “In the end, we used classic model tuning to select the best set of parameters.”
- Great job and great story! Once the edits are implemented I think this will be a really compelling journey!

**Comment by aloknsingh** [comment]
changes as per comments

**Comment by xwu0226** [comment]
@aloknsingh Just for your future reference. It would be ideal if @MadisonJMyers has a chance to do another check over your changes for accuracy and close the issue herself. Thanks!
